app/src/main/java/de/westnordost/streetcomplete/location/LocationStateButton.kt

```diff
@@ -11,7 +11,7 @@ import android.view.View
 import androidx.annotation.Keep
 import androidx.appcompat.widget.AppCompatImageButton
 import de.westnordost.streetcomplete.R
-import java.util.Arrays
+import de.westnordost.streetcomplete.location.LocationState.*
```

Comment:
`.*` :cry:

Reply:
OCD?

Reply:
Context: https://github.com/streetcomplete/StreetComplete/pull/3327#discussion_r717515209

Reply:
I know the context. I disagree (, but didn't care enough to comment on that pull request or suggest that Florian should change it).

The achievement constants/enum is exclusively used always in the exact same place, so the context is clear. The context is also clear for the `LocationStateButton` because it is literally a button that shows the `LocationState` and the method that returns a `LocationState` explicitly states that it returns a `LocationState`.

Basically, many things we do in Kotlin as normal will not work well if you are not using an IDE. Starting with lambdas whose parameter types are not annotated.

Reply:
Additionally/especially, wildcard imports make it much harder to do a source dive via GitHub.

I don't mind missing parameter types as much as wildcard imports. Usually they're not hard to figure out from context.

> many things we do in Kotlin as normal will not work well if you are not using an IDE.

It's the cost of using a language made by IDE developers, I guess :pensive:
But I don't have to like it :upside_down_face: